components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIFileUtil.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.apimgt.impl.utils;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.Enumeration;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+/**
+ * Util class for API file based operations.
+ */
+public class APIFileUtil {
+
+    public static final String WSDL_FILE_EXTENSION = "wsdl";
+    public static final String XSD_FILE_EXTENSION = "xsd";
+
+    private static final Logger log = LoggerFactory.getLogger(APIFileUtil.class);
+
+    public static String extractUploadedArchive(InputStream uploadedApiArchiveInputStream, String importedDirectoryName,
+            String apiArchiveLocation, String extractLocation) throws APIManagementException {
+        String archiveExtractLocation;
+        try {
+            // create api import directory structure
+            APIFileUtil.createDirectory(extractLocation);
+            // create archive
+            createArchiveFromInputStream(uploadedApiArchiveInputStream, apiArchiveLocation);
+            // extract the archive
+            archiveExtractLocation = extractLocation + File.separator + importedDirectoryName;
+            extractArchive(apiArchiveLocation, archiveExtractLocation);
+
+        } catch (APIManagementException e) {
+            APIFileUtil.deleteDirectory(extractLocation);
+            String errorMsg = "Error in accessing uploaded API archive";
```

Comment:
When does this error occur? which step?

Reply:
this happens when a user uploads a wsdl archive and then server extracts to the temp directory.